src/Validation/Validator.php

```diff
@@ -556,9 +573,13 @@ protected function _processRules($field, ValidationSet $rules, $data, $newRecord
                 continue;
             }
 
-            $errors[$name] = $message;
             if (is_string($result)) {
                 $errors[$name] = $result;
+            } elseif ($this->_useI18n) {
+                $args = $rule->get('pass');
+                $errors[$name] = __d($this->_validationDomain, $name, $this->_translateArgs($args));
```

Comment:
> I don't get why this is necessary.

The main advantage of this feature is DRYness. For e.g. I can just add translation "Valid email required" for "email" and avoid having to specify the message in each and every table where it's needed in every project. I can just keep these string is a separate translation domain file and copy it in a new project. I do the same currently in 2.x.

> No way to opt out of translations.

If you specify the 'message' explicitly when adding rule you opt out of this feature.

> Validation messages cannot be extracted with the other i18n tools.

Those who need the ability to be able to extract the messages can still set `'message' => __('my message')` when adding rules.